[BUG] Group by giving incorrect results

**Describe the bug**
 Group by giving incorrect results

**Steps/Code to reproduce bug**

```python

import cudf
import numpy as np
import pandas as pd
!conda list | grep cudf


count_df = cudf.DataFrame()
count_df['cat'] = np.random.randint(size=200_000,low=0, high=1000)
count_df['doc_id'] = np.random.randint(size=200_000, low=0, high=1000)
count_df['cnt'] = 1

cudf_res = count_df.groupby(['cat','doc_id'])['cnt'].count()
cudf_res.head(10)
```
Output:
```python

0    5         1
     5         2
918  419       1
0    5         2
     8         1
     5         2
918  419       1
0    5         2
     15        1
     5         2
Name: cnt, dtype: int32
```

The output seems to be visibily incorrect as we cant have groups with the same values being repeated, `0,5` in this case. 

**Expected behavior**
```
pd_res = count_df.to_pandas().groupby(['cat','doc_id'])['cnt'].count()
pd_res.head(10)
```

```
cat  doc_id
0    5         1
     8         1
     15        1
     17        1
     34        1
     48        3
     58        1
     61        1
     67        1
     77        1

```
Asserting values Result
```
pd.testing.assert_frame_equal(cudf_res.reset_index(drop=False).sort_values(by=['cat','doc_id','cnt']).reset_index().to_pandas(),
                              pd_res.reset_index(drop=False).sort_values(by=['cat','doc_id','cnt']).reset_index())

```

This leads to assertion error. 
```python
AssertionError: DataFrame.iloc[:, 0] are different
DataFrame.iloc[:, 0] values are different (98.96895 %)
```

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda

```
cudf                      0.9.0a                py37_2783    rapidsai-nightly
libcudf                   0.9.0a            cuda10.0_2783    rapidsai-nightly
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -169,6 +169,7 @@
 - PR #2567 Fix ColumnVector.fromScalar issues while dealing with null scalars
 - PR #2565 Orc reader: fix incorrect data decoding of int64 data types
 - PR #2577 Fix search benchmark compilation error by adding necessary header
+- PR #2604 Fix a bug in copying.pyx:_normalize_types that upcasted int32 to int64
 
 
 # cuDF 0.8.0 (27 June 2019)

--- a/python/cudf/cudf/bindings/copying.pyx
+++ b/python/cudf/cudf/bindings/copying.pyx
@@ -44,7 +44,7 @@ def clone_columns_with_size(in_cols, row_size):
 
 def _normalize_maps(maps, size):
     maps = columnops.as_column(maps).astype("int32")
-    maps = maps.binary_operator("mod", maps.normalize_binop_value(size))
+    maps = maps.binary_operator("mod", np.int32(size))
     maps = maps.data.mem
     return maps
 

